core/arch/arch_exports.h

```diff
@@ -220,6 +223,20 @@ typedef struct _local_state_extended_t {
 #    define SCRATCH_REG4 DR_REG_R4
 #    define SCRATCH_REG5 DR_REG_R5
 #    define SCRATCH_REG_LAST SCRATCH_REG5
+#elif defined(RISCV64)
+/* FIXME-RISCV: RFC: Are there any requirements on TLS/scratch registers?
+ * Aarch64 and X86 use a combination of argument, callee-saved and
+ * caller-saved/temporary regs. Can we use temporary registers or rather leave
+ * them to client code?
```

Comment:
Note to self: I'll use argument registers `aX` (alias to `x10-x17`) since those are consecutive and won't cause confusion as using `x1-x4` would (`x0` is a hard-wired-zero register).

Reply:
Done